components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/endpoint/registry/util/DefinitionValidator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.impl.endpoint.registry.util;
+
+import java.net.URL;
+
+/**
+ * An API Definition Validator Interface
+ */
+public interface DefinitionValidator {
+
+    boolean validate(URL definitionUrl);
+
+    boolean validate(byte[] definition);
+
+    boolean validate(String definition);
```

Comment:
Throw the Exceptions from the method signatures.

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/8618/commits/169ace7202aa557c7b6710f5bf9e481af4ff99d4